### ISSUE TITLE
On-demand formatter and JoinWith

### ISSUE THREAD

**Issue by kosmodisk** [propelorm/Propel2#847]
On-demand formatter cannot currently handle queries using joinwith:
https://github.com/propelorm/Propel2/blob/master/src/Propel/Runtime/Formatter/OnDemandFormatter.php#L50

```
throw new LogicException('OnDemandFormatter cannot hydrate related objects 
using a one-to-many relationship. Try removing with() from your query.');
```

But it would still be possible if query has been ordered by PK (all of its PKs) - then main model records are not scattered around resultset. 

I understand this idea may sound a little peculiar but it would be really handy when dealing with large resultsets (and where custom ordering is not important).

**Comment by marcj** [comment]
Yes, it'd be possible when we're ordering by PK then. Feel free to send us a pull-request, I'd love to merge it then.
